extension/src/plots/paths/collect.ts

```diff
@@ -126,31 +157,20 @@ const collectOrderedPath = (
   idx: number
 ): PlotPath[] => {
   const path = getPath(pathArray, idx)
+  const hasChildren = idx !== pathArray.length
 
   if (acc.some(({ path: existingPath }) => existingPath === path)) {
-    return acc.map(existing =>
-      existing.path === path
-        ? {
-            ...existing,
-            revisions: new Set([...existing.revisions, ...revisions])
-          }
-        : existing
-    )
+    return updateExistingPlotPath(acc, data, hasChildren, revisions, path)
```

Comment:
[I] In `updateExistingPlotPath` you are looping over the entire array again. This could be cut down to a single loop

Reply:
*I realise that this change is just expanding on what the old code was doing but we can still improve/update.